src/spn/algorithms/Inference.py

```diff
@@ -39,6 +39,21 @@ def prod_likelihood(node, children, data=None, dtype=np.float64):
     assert llchildren.dtype == dtype
     return np.prod(llchildren, axis=1).reshape(-1, 1)
 
+#added
+def max_log_likelihood(node, children, data=None, dtype=np.float64):
+    llchildren = np.concatenate(children, axis=1)
+    assert llchildren.dtype == dtype
+    #print("node and llchildren", (node, llchildren))
```

Comment:
you may want to use the python logging library, so a possible rewriting of this could be to use the logging library as opposed to prints.

Reply:
Sure. That's a good suggestion. I'll use the logging library for next revision